wodles/docker-listener/tests/test_docker_listener.py

```diff
@@ -0,0 +1,200 @@
+    mock_thread.assert_called_with(target=dl.listen)
+    dl.thread1.start.assert_called_once()
+
+
+@patch('DockerListener.docker.from_env', return_value=MagicMock())
+def test_DockerListener_check_docker_service(mock_env):
+    """Test check_docker_service verifies the docker service status."""
+    dl = docker_listener.DockerListener()
+    is_running = dl.check_docker_service()
+    mock_env.assert_called_once()
+    dl.client.ping.assert_called_once()
+    assert is_running
+
+
+@patch('DockerListener.docker.from_env', Exception)
+def test_DockerListener_check_docker_service_ko():
+    """Test check_docker_service returns False when the service is not running."""
+    dl = docker_listener.DockerListener()
+    assert not dl.check_docker_service()
+
+
+@patch('DockerListener.DockerListener.connect')
+@patch('DockerListener.DockerListener.send_msg')
+@patch('DockerListener.DockerListener.process')
+def test_DockerListener_listen(mock_process, mock_send, mock_connect):
+    """Test listen function process every event received from the client."""
+    event_list = [1, 2, 3]
+    dl = docker_listener.DockerListener()
+    dl.client = MagicMock()
+    dl.client.events.return_value = event_list
+    dl.listen()
+    mock_process.assert_has_calls([call(x) for x in event_list])
+    mock_send.assert_called_once()
+    mock_connect.assert_called_once()
+
+
+@patch('DockerListener.DockerListener.connect')
+@patch('DockerListener.DockerListener.send_msg')
+@patch('DockerListener.DockerListener.process')
+def test_DockerListener_listen_ko(mock_process, mock_send, mock_connect):
+    """Test listen raises an exception when is caught."""
+    dl = docker_listener.DockerListener()
+    dl.client = MagicMock()
+    dl.client.events.side_effect = Exception
+    with pytest.raises(Exception):
+        dl.listen()
+    mock_process.assert_not_called()
+    mock_send.assert_not_called()
+    mock_connect.assert_not_called()
+
+
+@patch('DockerListener.DockerListener.send_msg')
+def test_DockerListener_process(mock_send):
+    """Test process function sends the decoded events to the Wazuh socket."""
+    event = b"test"
+    dl = docker_listener.DockerListener()
+    dl.process(event)
+    mock_send.assert_called_with(event.decode("utf-8"))
+
+
+def test_DockerListener_format_msg():
+    """Test format_msg returns a dict with the expected key and values."""
+    msg = '{"test": "value"}'
+    dl = docker_listener.DockerListener()
+    result = dl.format_msg(msg)
+    assert isinstance(result, dict)
+    assert result.get("integration") == "docker"
+    assert result.get("docker") == json.loads(msg)
+
+
+@patch('DockerListener.socket.socket')
+def test_DockerListener_send_msg(mock_socket):
+    """Test send_msg sends the messages with the expected contents to the Wazuh socket."""
+    msg = '{"test": "value"}'
+
+    m = MagicMock()
+    mock_socket.return_value = m
+    dl = docker_listener.DockerListener()
+    dl.send_msg(msg)
+
```

Comment:
I would mock the `print` function too. You don't have to use `assert_called_with` as `json_msg` is tested below. Therefore, I would use just `assert_called_once`.

Reply:
This is not required as those prints are only intended for debugging purposes. Their output are not send to ossec.log. They will be replaced by logging in a future issue.